tests: Bluetooth: restructure `bsim_bt` directory

jori-nordic opened the pull request:
This commit separates each test by layer level.

Obviously the tests will run the whole stack, but they should be separated by which component/layer they intend on testing.

Thalley commented on tests/bluetooth/bsim/compile.sh:
```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
```
Since we are restructuring the bsim directory, would it make sense to also split this file into smaller compile files?

PavelVPV replied:
Would be nice. Then I don't need to comment out other test harnesses if I want to compile only the mesh ones.

jori-nordic replied:
The think that would make the most sense would be switching to twister for building and running bsim tests. Then we could do away with the ad-hoc shell scripts that we have.
But I thought that'd be too much scope, hence the rename-only PR.

hermabe replied:
This will become much nicer once we get the pytest plugin and twister v2 (ETA ??), then we can run and filter tests properly.

Thalley replied:
Looks good to me

jori-nordic committed:
test: Bluetooth: split compile.sh

Split into layers for now, we can have more granular splits later.

Signed-off-by: Jonathan Rico <jonathan.rico@nordicsemi.no>

jori-nordic committed:
tests: Bluetooth: restructure `bsim_bt` directory

This commit separates each test by layer level.

Obviously the tests will run the whole stack, but they should be separated
by which component/layer they intend on testing.

Signed-off-by: Jonathan Rico <jonathan.rico@nordicsemi.no>

aescolar commented on tests/bluetooth/bsim/compile.sh:
```diff
@@ -18,68 +18,9 @@ BOARD_ROOT="${BOARD_ROOT:-${ZEPHYR_BASE}}"
-app=tests/bluetooth/bsim/mesh conf_overlay=overlay_low_lat.conf compile &
-app=tests/bluetooth/bsim/mesh conf_overlay=overlay_pst.conf compile &
-app=tests/bluetooth/bsim/mesh conf_overlay=overlay_gatt.conf compile &
+${ZEPHYR_BASE}/tests/bluetooth/bsim/audio/compile.sh
```
better add an & here too, otherwise the audio one (which is a single app) is built in series with the others

jori-nordic replied:
good catch, messed up when reordering

aescolar reviewed (commented):
No fundamental concerns from my side :)

Thalley reviewed (dismissed):
Changes look good to me, but needs a rebase :)

jori-nordic commented:
> Changes look good to me, but needs a rebase :)

ugh, again. Updated now.
